File: service-server/src/main/java/com/rocketdan/serviceserver/domain/event/reward/Reward.java
package com.rocketdan.serviceserver.domain.event.reward;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.Getter;
import lombok.NoArgsConstructor;

import javax.persistence.*;

@Entity
@Getter
@NoArgsConstructor
public class Reward {
    @Id
    @JsonIgnore
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private Long level;

    private Integer category;

    @Column(nullable = false)
    private String name;

    private String image;

    private Integer price;

    private Integer count;

    public Reward(Long level, Integer category, String name, String image, Integer price, Integer count) {
        this.level = level;
        this.category = category;
        this.name = name;
        this.image = image;
        this.price = price;
        this.count = count;
    }
}

File: web-ui/lib/models/reward.dart
import 'reward_category.dart';

class Reward {
  final String name;
  final String imgPath;
  final int price;
  final int count;
  final int level;
  final RewardCategory category;

  String get getImgPath => imgPath;
  Reward(
      {required this.name,
      required this.imgPath,
      required this.price,
      required this.count,
      required this.level,
      required this.category});

  factory Reward.fromJson(Map<String, dynamic> json) {
    return Reward(
        name: json['name'],
        imgPath: json['image'],
        category: RewardCategory.values[json['category']],
        price: json['price'],
        count: json['count'],
        level: json['level']);
  }
}
